app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -968,11 +969,30 @@ class HomeFragment : Fragment() {
                                 text = resources.getString(R.string.wallpaper_select_error_snackbar_message),
                             )
                         }
+                        // If setting a wallpaper failed reset also the contrasting text color.
+                        requireContext().settings().currentWallpaperTextColor = 0L
                         lastAppliedWallpaperName = Wallpaper.defaultName
                     }
                 }
             }
         }
+        // Logo color should be updated in all cases.
+        applyWallpaperTextColor()
+    }
+
+    /**
+     * Apply a color better contrasting with the current wallpaper to the Fenix logo and private mode switcher.
+     */
+    private fun applyWallpaperTextColor() {
+        if (shouldEnableWallpaper()) {
```

Comment:
nit: This shouldn't strictly be necessary because it is checked in `applyWallpaper` as well but I'm fine with keep it if we want the extra assurance

Reply:
Will remove it.